cmake/extensions.cmake

```diff
@@ -383,6 +403,8 @@ endfunction()
 #
 # zephyr_library versions of normal CMake target_<func> functions
 #
+# New code is encouraged to use zephyr_library_cmake_target(), then the
+# usual CMake target_<func> functions, directly. See #8439.
```

Comment:
We should not deprecate zephyr_library_<func> in favor of target_<func>.

#8439 has some good points, but it is not merged, and it is not canon.

Wrapping the target_<func> API has several benefits, for instance we want
zephyr_library_compile_options to have precedence over zephyr_compile_options.

https://github.com/zephyrproject-rtos/zephyr/pull/16843/commits/3e7db9fc196992dcea1662c3b67b08bdd65341f4#diff-e64e7d3f59d40d3325fa7a458c531dd3L409

And many others. It can not be deprecated offhandedly like this.

Reply:
> Wrapping the target_ API has several benefits [...] It can not be deprecated offhandedly like this.

So no "encouragement" to stop using the `zephyr_` layer of indirection but is OK not to use it in a particular case like this one?
In other words is @mbolivar's new `zephyr_library_cmake_target()` "unwrapper" still OK in situations like this one or is just wrong in any case right now?

If the new "unwrapper" is wrong then should it be `target_link_libraries(${ZEPHYR_CURRENT_LIBRARY}"  PRIVATE kernel)`? Some new `zephyr_library_link_libraries_PRIVATE(...)` wrapper to be added? Other? To be determined?

Sorry to insist in an obsolete PR but this seems to me like a basic, one-line `zephyr_library_` question for a pretty simple use case. If it's actually complicated then please just say so and I'll create a new github issue for this question specifically.

Reply:
@marc-hb Sebastian is on vacation, so I'll try to answer on his behalf.

My understand was that a particular case like this should be avoided. There is a general pattern for how libraries are added and we should use that as much as we can.

If you need a zephyr_link_private then that should probably be added, there is already a zephyr_link_interface (although it is not in use anymore, so maybe that should be deleted).
I'm curious though, why do we need a PRIVATE version, I'm afraid I don't really understand the difference right now.
If you think PRIVATE is better maybe zephyr_library_link_libraries should be changed to use PRIVATE if possible.

If you wish to follow up on this, please create new github issue and assign Sebastian there.